gst-libs/gst/androidjni/gstjniamcutils.c

```diff
@@ -0,0 +1,294 @@
+  gst_amc_format_set_feature_enabled (format,
+      GST_AMC_MEDIA_FORMAT_TUNNELED_PLAYBACK, TRUE);
+
+  codec_list = gst_jni_media_codec_list_new ();
+  if (codec_list == NULL) {
+    GST_ERROR ("Could not get codec list");
+    goto done;
+  }
+  codec_name =
+      gst_jni_media_codec_list_find_decoder_for_format (codec_list, format);
+
+done:
+  if (format != NULL) {
+    gst_amc_format_free (format);
+  }
+  if (codec_list != NULL) {
+    gst_jni_media_codec_list_free (codec_list);
+  }
+  return codec_name;
+}
+
+
+gchar *
+gst_jni_amc_decoder_to_gst_plugin_name (gchar * codec_name)
+{
+#define PREFIX_LEN 10
+  gchar *element_name;
+  gint i, k;
+  gint codec_name_len;
+  const gchar *prefix = "amcviddec-";
+
+  // This is copied from gstamc.c to get the element name from the codec name
+  codec_name_len = strlen (codec_name);
+  element_name = g_new0 (gchar, PREFIX_LEN + strlen (codec_name) + 1);
+  memcpy (element_name, prefix, PREFIX_LEN);
+
+  for (i = 0, k = 0; i < codec_name_len; i++) {
+    if (g_ascii_isalnum (codec_name[i])) {
+      element_name[PREFIX_LEN + k++] = g_ascii_tolower (codec_name[i]);
+    }
+    /* Skip all non-alnum chars */
+  }
+#undef PREFIX_LEN
+
+  return element_name;
+}
+
+GList *
+gst_jni_amc_get_decoders_with_feature (GstCaps * caps, const gchar * feature)
+{
+  GList *list_ret = NULL;
+  GstJniMediaCodecList *codec_list = NULL;
+
+  jobjectArray jcodec_infos;
+  jint codec_count, i;
+
+  jobject codec_info = NULL;
+  jclass codec_info_class = NULL;
+  jmethodID get_capabilities_for_type_id;
+  jmethodID get_name_id;
+  jmethodID is_encoder_id;
+
+  jobject capabilities = NULL;
+  jclass capabilities_class = NULL;
+  jmethodID is_feature_supported;
+
+  jstring feature_jstring = NULL;
+  jstring type_jstring = NULL;
+
+  gchar *type = gst_jni_amc_video_caps_to_mime (caps);
+  JNIEnv *env = gst_jni_get_env ();
+
+  codec_info_class = (*env)->FindClass (env, "android/media/MediaCodecInfo");
+  if (!codec_info_class) {
+    GST_ERROR ("Can't find android/media/MediaCodecInfo class");
+    goto next_codec;
+  }
+
+  capabilities_class =
+      (*env)->FindClass (env, "android/media/MediaCodecInfo$CodecCapabilities");
+  if (!capabilities_class) {
+    GST_ERROR
+        ("Can't find android/media/MediaCodecInfo$CodecCapabilities class");
+    goto next_codec;
+  }
+  get_capabilities_for_type_id =
+      (*env)->GetMethodID (env, codec_info_class, "getCapabilitiesForType",
+      "(Ljava/lang/String;)Landroid/media/MediaCodecInfo$CodecCapabilities;");
+
+  get_name_id =
+      (*env)->GetMethodID (env, codec_info_class, "getName",
+      "()Ljava/lang/String;");
+
+  is_encoder_id =
+      (*env)->GetMethodID (env, codec_info_class, "isEncoder", "()Z");
+
+  is_feature_supported =
+      (*env)->GetMethodID (env, capabilities_class, "isFeatureSupported",
+      "(Ljava/lang/String;)Z");
+
+  GST_ERROR
+      ("methods: get_capabilities_for_type_id %d\tget_name_id %d\tis_encoder_id %d\tis_feature_supported %d",
```

Comment:
do we really need this trace? this seems to be a "development one"

Reply:
Of course it is not critical, we can remove it or set DEBUG level.